src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -1189,7 +1205,7 @@ private void SendPreLoginHandshake(byte[] instanceName, bool encrypt, bool clien
                         switch (_encryptionOption & EncryptionOptions.OPTIONS_MASK)
                         {
                             case (EncryptionOptions.ON):
-                                if (serverOption == EncryptionOptions.NOT_SUP)
+                                if ((serverOption & EncryptionOptions.OPTIONS_MASK) != EncryptionOptions.ON)
```

Comment:
Here too I feel it's same as before as there are only 2 possibilities here: ON or NOT_SUP, condition can be left untouched. Options Mask case can be taken care when it's enabled, need not be modified in this PR.